### ISSUE TITLE
[Question] Different output for same code in pipeline?

### ISSUE THREAD

**Issue by codeOfJannik** [dart-code-checker/dart-code-metrics#842]
If I locally run `flutter pub global run dart_code_metrics:metrics analyze --fatal-style --fatal-performance --fatal-warnings lib` with rules in the analysis_options.yaml in place, I don't receive warnings anymore since I have resolved all of them. But when the same command is executed inside a step of the bitbucket-pipeline.yaml I receive 46 warnings, most of them _avoid-dynamic_.

When I check the linked line of code there's no dynamic type used. That are just lines like: `class Museum with EquatableMixin {`

Why is there this difference in the pipeline? Is there anything I can adjust to get the command running properly in the pipeline?

**Comment by incendial** [comment]
@codeOfJannik could you share your `analysis_options.yaml` and  `bitbucket-pipeline.yaml` Dart Code Metrics configs?

**Comment by codeOfJannik** [comment]
**analysis-options.yaml**
```yaml
include: package:flutter_lints/flutter.yaml

analyzer:
  plugins:
    - dart_code_metrics
  exclude:
    - lib/generated/**
    - plugins/**
  errors:
    missing_required_param: warning
    parameter_assignments: warning
    missing_return: error
    todo: ignore

dart_code_metrics:
  metrics:
    - cyclomatic-complexity: 20
    - halstead-volume: 150
    - lines-of-code: 100
    - maintainability-index: 50
    - maximum-nesting-level: 5
    - number-of-methods: 10
    - number-of-parameters: 6
    - source-lines-of-code: 50
    - technical-debt:
        threshold: 1
        todo-cost: 161
        ignore-cost: 320
        ignore-for-file-cost: 396
        as-dynamic-cost: 322
        deprecated-annotations-cost: 37
        file-nullsafety-migration-cost: 41
        unit-type: "USD"
    - weight-of-class: 0.33
  rules:
    # Common
    - avoid-collection-methods-with-unrelated-types
    - avoid-dynamic:
        exclude:
          - test/**
    - avoid-global-state
    - avoid-late-keyword:
        exclude:
          - test/**
          - lib/main.dart
    - avoid-missing-enum-constant-in-map
    - avoid-nested-conditional-expressions:
        acceptance-level: 2
    - avoid-non-null-assertion
    - avoid-unnecessary-type-assertions
    - avoid-unnecessary-type-casts
    - avoid-unrelated-type-assertions
    - avoid-unused-parameters
    - binary-expression-operand-order
    - double-literal-format
    - format-comment
    - member-ordering:
      alphabetize: true
      order:
        - public-fields
        - private-fields
        - constructors
    - newline-before-return
    - no-boolean-literal-compare
    - no-empty-block
    - no-equal-arguments:
        ignored-parameters:
          - height
          - width
    - no-equal-then-else
    - no-magic-number
    - no-object-declaration
    - prefer-async-await
    - prefer-commenting-analyzer-ignores
    - prefer-conditional-expressions
    - prefer-correct-identifier-length:
        exceptions:
          - id
          - ok
    - prefer-correct-type-name:
        max-length: 44
    - prefer-first
    - prefer-last
    - prefer-immediate-return
    - prefer-match-file-name:
        exclude:
          - test/**
    - prefer-moving-to-variable:
        exclude:
          - test/**
    - prefer-trailing-comma
    # Flutter specific
    - always-remove-listener
    - avoid-border-all
    - avoid-returning-widgets
    - avoid-unnecessary-setstate
    - avoid-wrapping-in-padding
    - prefer-const-border-radius
    - prefer-extracting-callbacks
    - prefer-single-widget-per-file:
        ignore-private-widgets: true

  anti-patterns:
    - long-method
    - long-parameter-list

  metrics-exclude:
    - test/**

```

**bitbucket-pipelines.yaml**
```yaml
image: cirrusci/flutter:latest

pipelines:
  pull-requests:
    "**":
      - step:
          name: "Dependencies"
          caches:
            - flutter-pub-cache
          script:
            - git submodule update --recursive --init
            - flutter pub get
            - flutter packages pub global activate intl_utils
            - flutter pub global run intl_utils:generate
          artifacts:
            - lib/generated/**
            - plugins/**
            - .dart_tool/**
      - step:
          name: "Static code analysis"
          script:
            - flutter pub global activate dart_code_metrics
            - flutter pub global run dart_code_metrics:metrics analyze --fatal-style --fatal-performance --fatal-warnings lib
            # - flutter pub run dart_code_metrics:metrics check-unused-code lib
            # - flutter pub run dart_code_metrics:metrics check-unused-files lib
            # - flutter pub run dart_code_metrics:metrics check-unused-l10n lib
      - parallel:
          - step:
              name: "Build Android"
              caches:
                - gradle
                - gradlewrapper
                - android-sdk
                - flutter-pub-cache
              script:
                - flutter build apk --flavor dev -t lib/main_dev.dart
          - step:
              name: "Unit Tests"
              caches:
                - flutter-pub-cache
              script:
                - flutter test test/unit_tests
          - step:
              name: "Widget Tests"
              caches:
                - flutter-pub-cache
              script:
                - flutter test test/widget_tests/
definitions:
  caches:
    gradlewrapper: ~/.gradle/wrapper
    android-sdk: /opt/android-sdk-linux
    flutter-pub-cache: ~/.pub-cache

```

**Comment by dkrutskikh** [comment]
@codeOfJannik thank you for info, could you please attach output from cli and bitbucket report ?

**Comment by codeOfJannik** [comment]
**CLI output:**
```
~/... ❯ flutter pub global activate dart_code_metrics                                                                                             09:19:29
Package dart_code_metrics is currently active at version 4.15.0.
Resolving dependencies...
+ _fe_analyzer_shared 36.0.0 (39.0.0 available)
+ analyzer 3.3.1 (4.0.0 available)
+ analyzer_plugin 0.9.0
+ ansicolor 2.0.1
+ args 2.3.1
+ async 2.9.0
+ charcode 1.3.1
+ collection 1.16.0
+ convert 3.0.1
+ crypto 3.0.2
+ csslib 0.17.1
+ dart_code_metrics 4.15.0
+ dart_style 2.2.3
+ file 6.1.2
+ glob 2.0.2
+ html 0.15.0
+ meta 1.7.0
+ package_config 2.0.2
+ path 1.8.1
+ petitparser 5.0.0
+ pub_semver 2.1.1
+ source_span 1.9.0
+ string_scanner 1.1.0
+ term_glyph 1.2.0
+ typed_data 1.3.0
+ watcher 1.0.1
+ xml 6.0.1
+ yaml 3.1.0
Building package executables...
Built dart_code_metrics:metrics.
Installed executable metrics.
Warning: Pub installs executables into $HOME/.pub-cache/bin, which is not on your path.
You can fix that by adding this to your shell's config file (.bashrc, .bash_profile, etc.):

  export PATH="$PATH":"$HOME/.pub-cache/bin"

Activated dart_code_metrics 4.15.0.
~/... ❯ flutter pub global run dart_code_metrics:metrics analyze --fatal-style --fatal-performance --fatal-warnings lib                        4s 09:20:40
✔ no issues found!
```

**bitbucket output:**
```

lib/providers/provider/firebase_providers/storage_provider.dart:
WARNING Avoid using dynamic type.
        lib/providers/provider/firebase_providers/storage_provider.dart:84:5
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
WARNING Avoid using dynamic type.
        lib/providers/provider/firebase_providers/storage_provider.dart:88:7
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
WARNING Avoid using dynamic type.
        lib/providers/provider/firebase_providers/storage_provider.dart:130:30
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
WARNING Avoid using dynamic type.
        lib/providers/provider/firebase_providers/storage_provider.dart:149:7
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/providers/provider/shared_pref_provider.dart:
WARNING Avoid using dynamic type.
        lib/providers/provider/shared_pref_provider.dart:40:3
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/providers/provider/permission_provider.dart:
WARNING Avoid using dynamic type.
        lib/providers/provider/permission_provider.dart:86:9
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
WARNING Avoid using dynamic type.
        lib/providers/provider/permission_provider.dart:86:36
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/providers/provider/places_api_provider.dart:
WARNING Avoid using dynamic type.
        lib/providers/provider/places_api_provider.dart:42:3
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
WARNING Avoid using dynamic type.
        lib/providers/provider/places_api_provider.dart:44:22
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
WARNING Avoid using dynamic type.
        lib/providers/provider/places_api_provider.dart:58:7
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
WARNING Avoid using dynamic type.
        lib/providers/provider/places_api_provider.dart:62:7
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/ui/onboarding/onboarding_page_slider.dart:
WARNING Avoid using dynamic type.
        lib/ui/onboarding/onboarding_page_slider.dart:113:35
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/ui/shared/item_scanner.dart:
WARNING Avoid using dynamic type.
        lib/ui/shared/item_scanner.dart:160:23
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
WARNING Avoid using dynamic type.
        lib/ui/shared/item_scanner.dart:160:40
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/model/museum.dart:
WARNING Avoid using dynamic type.
        lib/model/museum.dart:37:14
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/model/item.dart:
WARNING Avoid using dynamic type.
        lib/model/item.dart:39:12
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/model/places_api/places_api_data.dart:
WARNING Avoid using dynamic type.
        lib/model/places_api/places_api_data.dart:37:21
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/cubits/shared_preferences/shared_pref_state.dart:
WARNING Avoid using dynamic type.
        lib/cubits/shared_preferences/shared_pref_state.dart:35:32
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/cubits/shared_preferences/shared_pref_cubit.dart:
WARNING Avoid using dynamic type.
        lib/cubits/shared_preferences/shared_pref_cubit.dart:40:23
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/cubits/permissions/permission_state.dart:
WARNING Avoid using dynamic type.
        lib/cubits/permissions/permission_state.dart:35:32
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/cubits/permissions/location/location_permission_cubit.dart:
WARNING Avoid using dynamic type.
        lib/cubits/permissions/location/location_permission_cubit.dart:38:31
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/cubits/item/item_state.dart:
WARNING Avoid using dynamic type.
        lib/cubits/item/item_state.dart:4:26
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/cubits/item/item_cubit.dart:
WARNING Avoid using dynamic type.
        lib/cubits/item/item_cubit.dart:15:17
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
WARNING Avoid collection methods with unrelated types.
        lib/cubits/item/item_cubit.dart:49:11
        avoid-collection-methods-with-unrelated-types : https://dartcodemetrics.dev/docs/rules/common/avoid-collection-methods-with-unrelated-types
lib/cubits/dynamic_links/dynamic_link_cubit.dart:
WARNING Avoid using dynamic type.
        lib/cubits/dynamic_links/dynamic_link_cubit.dart:44:24
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
WARNING Avoid using dynamic type.
        lib/cubits/dynamic_links/dynamic_link_cubit.dart:78:34
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
WARNING Avoid using dynamic type.
        lib/cubits/dynamic_links/dynamic_link_cubit.dart:92:13
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/cubits/dynamic_links/dynamic_link_state.dart:
WARNING Avoid using dynamic type.
        lib/cubits/dynamic_links/dynamic_link_state.dart:34:33
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/cubits/museum/museum_cubit.dart:
WARNING Avoid using dynamic type.
        lib/cubits/museum/museum_cubit.dart:40:19
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/cubits/museum/museum_state.dart:
WARNING Avoid using dynamic type.
        lib/cubits/museum/museum_state.dart:39:28
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/cubits/artist/artist_state.dart:
WARNING Avoid using dynamic type.
        lib/cubits/artist/artist_state.dart:36:28
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
lib/cubits/artist/artist_cubit.dart:
WARNING Avoid using dynamic type.
        lib/cubits/artist/artist_cubit.dart:40:19
        avoid-dynamic : https://dartcodemetrics.dev/docs/rules/common/avoid-dynamic
pub finished with exit code 2
```

**Comment by incendial** [comment]
Available in 4.16.0! 🚀
